cognite/extractorutils/extraction_pipelines.py

```diff
@@ -0,0 +1,108 @@
+
+    Args:
+        extraction_pipeline_ext_id:
+        cognite_client:
+        logger:
+        heartbeat_waiting_time:
+        added_message:
+
+    Usage:
+        If you have a function named "extract_data(*args, **kwargs)" and want to connect it to an extraction
+        pipeline, you can use this decorator function as:
+        @add_extraction_pipeline(
+            extraction_pipeline_ext_id=<INSERT EXTERNAL ID>,
+            cognite_client=<INSERT COGNITE CLIENT OBJECT>,
+            logger=<INSERT LOGGER>,
+        )
+        def extract_data(*args, **kwargs):
+            <INSERT FUNCTION BODY>
+
+    Potential Improvements:
+    -- Refactor so this decorator share methods with the Extractor context manager in .base.py as they serve a similar
+    purpose
+    -- Change 'cognite_client.extraction_pipeline_runs' -> 'cognite_client.extraction_pipelines.runs'
+    when SDK is updated
+
+    """
+
+    cancellation_token: Event = Event()
+
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    def decorator_ext_pip(input_function):
+        @wraps(input_function)
+        def wrapper_ext_pip(*args, **kwargs):
+            ##############################
+            # Setup Extraction Pipelines #
+            ##############################
+            logger.info("Setting up Extraction Pipelines")
+
+            def _report_success() -> None:
+                message = f"Successful shutdown of function '{input_function.__name__}'. {added_message}"
+                cognite_client.extraction_pipeline_runs.create(  # cognite_client.extraction_pipelines.runs.create(
+                    ExtractionPipelineRun(external_id=extraction_pipeline_ext_id, status="success", message=message)
+                )
+
+            def _report_error(exception: Exception) -> None:
+                """
+                Called on an unsuccessful exit of the extractor
+                """
+                message = (
+                    f"Exception for function '{input_function.__name__}'. {added_message}:\n" f"{str(exception)[:1000]}"
+                )
+                cognite_client.extraction_pipeline_runs.create(
+                    ExtractionPipelineRun(external_id=extraction_pipeline_ext_id, status="failure", message=message)
+                )
+
+            def heartbeat_loop() -> None:
+                while not cancellation_token.is_set():
+                    cognite_client.extraction_pipeline_runs.create(
+                        ExtractionPipelineRun(external_id=extraction_pipeline_ext_id, status="seen")
+                    )
+
+                    cancellation_token.wait(heartbeat_waiting_time)
+
+            ##############################
+            # Run the extractor function #
+            ##############################
+            logger.info(f"Starting to run function: {input_function.__name__}")
+
+            print(cancellation_token.is_set())
```

Comment:
Avoid using `print` like this. If you need something logged you should use the logger instead, but I suspect this might be some debug printing that was just accidentally included?

Reply:
Pure mistake on my part. Thanks for pointing it out. Will fix.